pkg/supervisor/supervisor.go

```diff
@@ -130,14 +142,33 @@ func (s *supervisor) Start(ctx context.Context) {
 	s.reflectorCtl.Start(ctx)
 	defer events.Log("Stopped Supervisor")
 	for {
+		err := func() error {
+			// If the ledger latency is too much, temporarily stop uploading snapshots.
+			// We need to first catch up, or else we'll upload snapshots that are out-of-date
+			// which would put a significant amount of load on the exective because every new
+			// reflector will have to sync a potentially very large chunk of the DML ledger.
+			latency, err := s.reader.GetLedgerLatency(ctx)
+			if err != nil {
+				return err
+			}
+			isAcceptableLatency := s.minimumLedgerLatency > latency
+			stats.Set("is_acceptable_latency", isAcceptableLatency)
+			stats.Set("latency", latency)
```

Comment:
We already instrument ledger latency in the supervisor https://segment.datadoghq.com/dashboard/5hd-biq-s3m/ctlstore?from_ts=1571130225756&fullscreen_widget=362012904&live=true&tile_size=m&to_ts=1571144625756&fullscreen_section=overview so we can probably not worry about doing it here.

Reply:
Neat! Thanks